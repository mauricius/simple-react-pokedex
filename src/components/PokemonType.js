import React from 'react'

const PokemonType = (props) => (
  <span className="inline-block bg-grey-lighter border border-grey rounded-full px-3 py-1 text-sm font-semibold text-grey-darker mr-2">{props.type.type.name}</span>
)

export default PokemonType
