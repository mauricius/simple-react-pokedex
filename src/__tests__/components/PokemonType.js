import React from 'react'
import PokemonType from '../../components/PokemonType'
import renderer from 'react-test-renderer'

describe('test PokemonType component', function () {

  it('renders correctly', () => {
    const type = {
      type : {
        name: 'grass'
      }
    }

    const component = renderer
      .create(<PokemonType type={type} />)
      .toJSON()

    expect(component).toMatchSnapshot()
  });
});
