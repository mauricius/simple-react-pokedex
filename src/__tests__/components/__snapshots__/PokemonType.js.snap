// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PokemonType component renders correctly 1`] = `
<span
  className="inline-block bg-grey-lighter border border-grey rounded-full px-3 py-1 text-sm font-semibold text-grey-darker mr-2"
>
  grass
</span>
`;
